pytorch_lightning/callbacks/lr_logger.py

```diff
@@ -64,22 +70,36 @@ def on_train_start(self, trainer, pl_module):
         # Initialize for storing values
         self.lrs = {name: [] for name in names}
 
-    def on_train_batch_start(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
-        latest_stat = self._extract_lr(trainer, 'step')
-        if trainer.logger and latest_stat:
-            trainer.logger.log_metrics(latest_stat, step=trainer.global_step)
-
+    def on_batch_start(self, trainer, pl_module):
+        if self.logging_interval is None:
+            latest_stat = self._extract_lr(trainer, 'step')
+            if trainer.logger and latest_stat:
+                trainer.logger.log_metrics(latest_stat, step=trainer.global_step)
+        elif self.logging_interval == 'step':
+            latest_stat = self._extract_lr(trainer, 'any')
+            if trainer.logger and latest_stat:
+                trainer.logger.log_metrics(latest_stat, step=trainer.global_step)
+        else:
+            pass
+    
     def on_epoch_start(self, trainer, pl_module):
-        latest_stat = self._extract_lr(trainer, 'epoch')
-        if trainer.logger and latest_stat:
-            trainer.logger.log_metrics(latest_stat, step=trainer.global_step)
-
+        if self.logging_interval is None:
+            latest_stat = self._extract_lr(trainer, 'epoch')
+            if trainer.logger and latest_stat:
+                trainer.logger.log_metrics(latest_stat, step=trainer.current_epoch)
+        elif self.logging_interval == 'epoch':
+            latest_stat = self._extract_lr(trainer, 'any')
+            if trainer.logger and latest_stat:
+                trainer.logger.log_metrics(latest_stat, step=trainer.current_epoch)
+        else:
+            pass
```

Comment:
```suggestion
```
How about having a MisconfigurationException if `logging_interval` is not set to either None, 'step' or 'epoch'?